### PR TITLE
Add protected procedures

### PR DESCRIPTION
**Pull request by codetheweb**
Closes https://github.com/sfbrigade/compass/issues/52.

### CODE REVIEW

File: src/backend/auth/options.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import GoogleProvider from "next-auth/providers/google";
+import { createPersistedAuthAdapter } from "@/backend/auth/adapter";
+import { KyselyDatabaseInstance } from "../lib";
+
+export const getNextAuthOptions = (db: KyselyDatabaseInstance) => ({
```
Comment (codetheweb) [inline src/backend/auth/options.ts:5]: extracted out, nothing new here

File: src/backend/db/lib/seed.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { KyselyDatabaseInstance } from "@/backend/lib";
+
+export type SeedResult = Awaited<ReturnType<typeof seed>>;
```
Comment (amantri) [inline src/backend/db/lib/seed.ts:3]: What is the purpose of this file?

Comment (codetheweb) [reply src/backend/db/lib/seed.ts:3]: currently it's just for testing, but would be easy to set up as a dev script for some test data

can move it to the test folder for now

Comment (codetheweb) [reply src/backend/db/lib/seed.ts]: d85cfc0

File: src/backend/context.ts [hunk 3]
```diff
@@ -1,10 +1,51 @@
 import { CreateNextContextOptions } from "@trpc/server/adapters/next";
 import { getDb } from "backend/db/lib/get-db";
+import { Session, getServerSession } from "next-auth";
 import { Env } from "./lib/types";
+import { getNextAuthOptions } from "./auth/options";
 
-export const createContext = (options: CreateNextContextOptions) => {
+type Auth =
+  | {
```
Comment (amantri) [inline src/backend/context.ts:8]: OOC, what is the leading pipe on this line doing - adding null or undefined to `Auth` or it's syntactical sugar to line up the pipe on line 11?

**Review by amantri — APPROVED** [review]

File: src/backend/context.ts (hunk 3 above)
Comment (codetheweb) [reply src/backend/context.ts:8]: Put there by Prettier to line up the pipes 👍

**Pull request merged**